Makefile

```diff
@@ -57,6 +57,9 @@ fmt-check:
 fmt-fix:
 	sbt scalafmtAll scalafmtSbt
 
+fmt-fix-unused:
+	sbt "scalafix RemoveUnused"
```

Comment:
Would it be safe to run this under `fmt-fix` as well? I mean fmt-fix: `sbt scalafmtAll scalafmtSbt && sbt "scalafix RemoveUnused"`. I think there's a better chance we stick to using this if its part of the process we are already used to do.

Reply:
That's a good point. I did try putting it in the `fmt-fix` target at first, but it seemed to me resulted in a significant slowdown (by a factor of 3 or 4), due to the need to compile all the code to identify unused identifiers. I was worried that we might find that too costly.

Maybe I can take temperature on the tradeoff in today's meeting.

Reply:
In the meeting we all agreed that we'd like to check for unused values and imports in CI. The feelings on whether this should always be included in the format fix were a little less clear, but I think people were generally in favor of it.

Reply:
So, afaict, there's no straightforward way to fail CI when unused values/imports are found other then by enabling `-Yfatal-warnings`. I think we *should* do this, but as part of #1064, the prerequisite for which is fixing all of the compiler warnings. 

I have run a couple micro benchmarks, and for a no-op build, the cost of adding unused removal is in the order of 5 to 20 extra seconds on top of `fmt-fix`. That's pretty slow, but I think our current workflows are that this isn't run more than once or twice per PR anyhow. Hopefully we all have "format on save" and linting in our IDEs set up to catch most things before they make it to CI?

So I've added remove unused to the make target for now. We can always remove in followup if it proves too onerous.